### PR TITLE
UI link component

### PR DESCRIPTION
**Pull request by alston-hsu**

### CODE REVIEW

File: packages/UI/src/components/atoms/Link/index.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface LinkProps {
+  linkTo?: string;
+  linkText?: string;
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/index.ts:3]: @alston-hsu 
I think these should be: `href` and `text`

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  visited: theme.colors.LIGHT_BLUE_500,
+  hover: theme.colors.LIGHT_BLUE_200,
+  focus: theme.colors.BLUE_600,
+  focusBorder: theme.colors.LIGHT_BLUE_200,
+  active: theme.colors.BLUE_100,
+  disabled: theme.colors.GREYSCALE_GREY,
+};
+
+export const StyledLink = styled.a`
+  ${typographyStyles.pSmall}
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:26]: @alston-hsu 
I think the typography variant of link isn't yet defined so we have to create these [Typography variants](https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/Dev-Launchers-Style-Guide?node-id=1044%3A1054&t=O439lgOvsitp3B4S-4)
image
`typographyStyles.pSmall` is a bit different than them

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  color: ${lightColor.default};
+  @media only screen and (min-width: 62.5rem) {
+    ${typographyStyles.p}
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:30]: @alston-hsu 
I think the typography variant of link isn't yet defined so we have to create these [Typography variants](https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/Dev-Launchers-Style-Guide?node-id=1044%3A1054&t=O439lgOvsitp3B4S-4)
image
`typographyStyles.p` is a bit different than them

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  &:visited {
+    font-weight: 400;
+    color: ${lightColor.visited};
+    @media only screen and (min-width: 62.5rem) {
+      ${typographyStyles.p}
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:37]: @alston-hsu 
I think the typography variant of link isn't yet defined so we have to create these [Typography variants](https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/Dev-Launchers-Style-Guide?node-id=1044%3A1054&t=O439lgOvsitp3B4S-4)
image
`typographyStyles.p` is a bit different than them

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    }
+  }
+
+  &:hover {
+    font-weight: 600;
+    color: ${lightColor.hover};
+    @media only screen and (min-width: 62.5rem) {
+      ${typographyStyles.p}
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:45]: @alston-hsu 
I think the typography variant of link isn't yet defined so we have to create these [Typography variants](https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/Dev-Launchers-Style-Guide?node-id=1044%3A1054&t=O439lgOvsitp3B4S-4)
image
`typographyStyles.p` is a bit different than them

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    }
+  }
+
+  &:focus {
+    font-weight: 400;
+    color: ${lightColor.focus};
+    border: dotted 0.4px ${lightColor.focusBorder};
+    outline: 1px dashed black;
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:53]: @alston-hsu why are we doing border and outline at the same time here?

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    border-radius: 4px;
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:54]: @alston-hsu 
please use `packages/UI/src/components/atoms/Radius/StyledRadius.ts` file to for this radius.

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    background-color: ${lightColor.focusBorder};
+    @media only screen and (min-width: 62.5rem) {
+      ${typographyStyles.p}
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:57]: @alston-hsu 
I think the typography variant of link isn't yet defined so we have to create these [Typography variants](https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/Dev-Launchers-Style-Guide?node-id=1044%3A1054&t=O439lgOvsitp3B4S-4)
image
`typographyStyles.p` is a bit different than them

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  &:focus {
+    font-weight: 400;
+    color: ${lightColor.focus};
+    border: dotted 0.4px ${lightColor.focusBorder};
+    outline: 1px dashed black;
+    border-radius: 4px;
+    background-color: ${lightColor.focusBorder};
+    @media only screen and (min-width: 62.5rem) {
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:56]: @alston-hsu 
The theme already has breakpoints can you please use the equivalent of this one from the theme?

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+      ${typographyStyles.p}
+    }
+  }
+
+  &:active {
+    font-weight: 400;
+    color: ${lightColor.active};
+    @media only screen and (min-width: 62.5rem) {
+      ${typographyStyles.p}
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:65]: @alston-hsu 
I think the typography variant of link isn't yet defined so we have to create these [Typography variants](https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/Dev-Launchers-Style-Guide?node-id=1044%3A1054&t=O439lgOvsitp3B4S-4)
image
`typographyStyles.p` is a bit different than them

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    }
+  }
+
+  &:disabled {
+    font-style: italic;
+    font-weight: 300;
+    color: ${lightColor.disabled};
+    @media only screen and (min-width: 62.5rem) {
+      ${typographyStyles.p}
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:74]: @alston-hsu 
I think the typography variant of link isn't yet defined so we have to create these [Typography variants](https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/Dev-Launchers-Style-Guide?node-id=1044%3A1054&t=O439lgOvsitp3B4S-4)
image
`typographyStyles.p` is a bit different than them

File: packages/UI/src/components/atoms/Link/StyledLink.ts [hunk 12]
```diff
@@ -0,0 +1,77 @@
+import styled from 'styled-components';
+import theme from '../../../styles/theme';
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/StyledLink.ts:2]: @alston-hsu 
importing the theme is highly discouraged, please use this approach https://styled-components.com/docs/advanced#function-themes

File: packages/UI/src/components/atoms/Link/Link.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import type { Story, Meta } from '@storybook/react';
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/Link.stories.tsx:1]: @alston-hsu 
`ComponentStory` and `ComponentMeta` should be used here instead. please check other components to see 
the way we used those types

File: packages/UI/src/components/atoms/Link/Link.tsx [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import type { FC } from 'react';
+import { StyledLink } from './StyledLink';
+import type { LinkProps } from './index';
+
+const Link: FC<LinkProps> = ({ linkTo, linkText }) => {
```
Comment (Enjoy2Live) [inline packages/UI/src/components/atoms/Link/Link.tsx:5]: let's spread the rest of the props as follows `const Link: FC<LinkProps> = {linkTo, linkText, ...props})`
and use it on the StyledLink like this: `<StyledLink href={linkTo} {...props}>{linkText}</StyledLink>`

**Review by Enjoy2Live — CHANGES_REQUESTED** [review]
Looks pretty good, we need to complete those requested changes before we can merge.
They're just coding style related as the component looks almost identical to the Figma.

Thanks @alston-hsu :rocket:

**Review by alston-hsu — COMMENTED** [review]
changes added

**Review by Enjoy2Live — APPROVED** [review]

**Pull request merged**